pkg/pipelines/initialise.go

```diff
@@ -0,0 +1,247 @@
+			Verbs:     []string{"get", "patch"},
+		},
+	}
+)
+
+const (
+	pipelineDir       = "pipelines"
+	cicdDir           = "cicd-environment"
+	envsDir           = "envs"
+	baseDir           = "base"
+	appsDir           = "apps"
+	kustomize         = "kustomization.yaml"
+	namespacesPath    = "01-namespaces/cicd-environment.yaml"
+	rolesPath         = "02-rolebindings/pipeline-service-role.yaml"
+	rolebindingsPath  = "02-rolebindings/pipeline-service-rolebinding.yaml"
+	tasksPath         = "03-tasks/deploy-from-source-task.yaml"
+	ciPipelinesPath   = "04-pipelines/ci-dryrun-from-pr-pipeline.yaml"
+	cdPipelinesPath   = "04-pipelines/cd-deploy-from-push-pipeline.yaml"
+	prBindingPath     = "05-bindings/github-pr-binding.yaml"
+	pushBindingPath   = "05-bindings/github-push-binding.yaml"
+	prTemplatePath    = "06-templates/ci-dryrun-from-pr-template.yaml"
+	pushTemplatePath  = "06-templates/cd-deploy-from-push-template.yaml"
+	eventListenerPath = "07-eventlisteners/cicd-event-listener.yaml"
+	routePath         = "08-routes/github-webhook-event-listener.yaml"
+)
+
+// Initialise function will initialise the gitops directory
+func Initialise(o *InitialiseParameters) error {
+
+	if !o.SkipChecks {
+		installed, err := checkTektonInstall()
+		if err != nil {
+			return fmt.Errorf("failed to run Tekton Pipelines installation check: %w", err)
+		}
+		if !installed {
+			return errors.New("failed due to Tekton Pipelines or Triggers are not installed")
+		}
+	}
+
+	namespaces := namespaceNames(o.Prefix)
+
+	gitopsName := getGitopsRepoName(o.GitOpsRepo)
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	// check if the gitops dir already exists
+	exists, _ := isExisting(gitopsPath)
+	if exists {
+		return fmt.Errorf("%s already exists at %s", gitopsName, gitopsPath)
+	}
+
+	// create gitops pipeline
+	files := createPipelineResources(namespaces, o.GitOpsRepo, o.Prefix)
+
+	pipelinesPath := getPipelinesDir(gitopsPath, o.Prefix)
+
+	fileNames, err := writeResources(pipelinesPath, o.Prefix, files)
+	if err != nil {
+		return err
+	}
+
+	sort.Strings(fileNames)
+	// kustomize file should refer all the pipeline resources
+	if err := addKustomize("resources", fileNames, filepath.Join(pipelinesPath, kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	if err := addKustomize("bases", []string{"./pipelines"}, filepath.Join(getCICDDir(gitopsPath, o.Prefix), kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	if err := addKustomize("bases", []string{}, filepath.Join(gitopsPath, envsDir, baseDir, kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	fmt.Println("\nGitops initialised")
+	return nil
+}
+
+func getCICDDir(path, prefix string) string {
+	return filepath.Join(path, envsDir, addPrefix(prefix, cicdDir))
+}
+
+func createPipelineResources(namespaces map[string]string, gitopsRepo, prefix string) map[string][]interface{} {
+
+	// key: path of the resource
+	// value: YAML content of the resource
+	outputs := make(map[string][]interface{}, 0)
```

Comment:
You can shorten this to

```suggestion
	outputs := map[string][]interface{}{}
```